### PR TITLE
Face detection Open cv

### PR DESCRIPTION
**Pull request by ryuk156**
issue : https://github.com/ankitdobhal/Awesome-Python-Scripts/issues/93

About Pull request :
This pr consist of face detection in an video if a face is detected in an video the green rectangle will be drawn, it uses pre-trained face detect classifier

Demo:

image

### CODE REVIEW

**Comment by ryuk156** [comment]
@ankitdobhal  please review and merge this pr and  you have merged my car detection pr but that consist of car_detection.py  file but now after you merge that pr somehow file is deleted merge this pr I have included that file in this pr i think when you are making commits the file was deleted

File: Image-Processing/Face_detection/Face_detection.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import cv2    ##Import OpenCv
+
+face = cv2.CascadeClassifier('C:/Users/91976/Desktop/haarcascade_frontalface_default.xml') ##'haarcascade_frontalface_default.xml' an opencv classifier for face detection 
+
```
Comment (disha2sinha) [inline Image-Processing/Face_detection/Face_detection.py]: Remove new line

File: Image-Processing/Face_detection/Face_detection.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+cap = cv2.VideoCapture(0)   ##capturevideo by webcam or your laptop default cam for webcame  => 1 and for defalut laptop camera => 2
+
+while True:
+    sucess,img=cap.read()
+
```
Comment (disha2sinha) [inline Image-Processing/Face_detection/Face_detection.py]: Remove new line

File: Image-Processing/Face_detection/Face_detection.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import cv2    ##Import OpenCv
+
+face = cv2.CascadeClassifier('C:/Users/91976/Desktop/haarcascade_frontalface_default.xml') ##'haarcascade_frontalface_default.xml' an opencv classifier for face detection 
+
+cap = cv2.VideoCapture(0)   ##capturevideo by webcam or your laptop default cam for webcame  => 1 and for defalut laptop camera => 2
+
+while True:
+    sucess,img=cap.read()
```
Comment (disha2sinha) [inline Image-Processing/Face_detection/Face_detection.py]: put space after comma
put space around(after and before) =

File: Image-Processing/Face_detection/Face_detection.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    gray=cv2.cvtColor(img,cv2.COLOR_BGR2GRAY) ##before detecting  face you should convert img or video into gray image
+    faces=face.detectMultiScale(gray,1.1,5)
```
Comment (disha2sinha) [inline Image-Processing/Face_detection/Face_detection.py]: put space around =
put space after comma

File: Image-Processing/Face_detection/Face_detection.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    for ( x , y , w , h) in faces:
+        cv2.rectangle(img,( x , y ),( x*w , y*h ),(255,0,0),2)  ##will create the rectangle where face is detected
```
Comment (disha2sinha) [inline Image-Processing/Face_detection/Face_detection.py:12]: remove space before comma
put space after comma

File: Image-Processing/Face_detection/Face_detection.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    cv2.imshow('img',img) 
+    k=cv2.waitKey(30) & 0xff
+    if k==27:
```
Comment (disha2sinha) [inline Image-Processing/Face_detection/Face_detection.py]: remove space before comma
put space after comma
put space around = and ==

**Review by disha2sinha — CHANGES_REQUESTED** [review]
Remove car detection from this PR

**Comment by ryuk156** [comment]
@ankitdobhal @disha2sinha   merge this all the changes are done perfectly I have also removed car_detection.py

**Review by disha2sinha — APPROVED** [review]

**Pull request merged**